test/e2e-tests.sh

```diff
@@ -127,87 +223,163 @@ function add_trap() {
-  kubectl delete -f "${STRIMZI_INSTALLATION_CONFIG}" -n kafka
-  kubectl delete namespace kafka
+  kubectl delete -f ${KAFKA_INSTALLATION_CONFIG} -n "${STRIMZI_KAFKA_NAMESPACE}"
+  kubectl delete -f "${STRIMZI_INSTALLATION_CONFIG}" -n "${STRIMZI_KAFKA_NAMESPACE}"
+  kubectl delete namespace "${STRIMZI_KAFKA_NAMESPACE}"
+}
+
+# Installs the resources necessary to test the consolidated channel, runs those tests, and then cleans up those resources
+function test_consolidated_channel() {
+  # Test the consolidated channel
+  echo "Testing the consolidated channel"
+  install_consolidated_channel_crds || return 1
+  install_consolidated_sources_crds || return 1
+
+  go_test_e2e -timeout=40m -parallel=${TEST_PARALLEL} ./test/e2e -channels=messaging.knative.dev/v1alpha1:KafkaChannel,messaging.knative.dev/v1beta1:KafkaChannel  || fail_test
+  go_test_e2e -timeout=5m -parallel=${TEST_PARALLEL} ./test/conformance -channels=messaging.knative.dev/v1beta1:KafkaChannel -sources=sources.knative.dev/v1beta1:KafkaSource || fail_test
+
+  uninstall_sources_crds || return 1
+  uninstall_channel_crds || return 1
 }
 
-initialize $@ --skip-istio-addon
+# Installs the resources necessary to test the distributed channel, runs those tests, and then cleans up those resources
+function test_distributed_channel() {
+  # Test the distributed channel
+  echo "Testing the distributed channel"
+  install_distributed_channel_crds || return 1
+  install_distributed_sources_crds || return 1
+
+  # TODO: Enable v1alpha1 testing once we have the auto-converting webhook in our config/yaml
+  go_test_e2e -timeout=40m -parallel=${TEST_PARALLEL} ./test/e2e -channels=messaging.knative.dev/v1beta1:KafkaChannel  || fail_test
+  go_test_e2e -timeout=5m -parallel=${TEST_PARALLEL} ./test/conformance -channels=messaging.knative.dev/v1beta1:KafkaChannel -sources=sources.knative.dev/v1beta1:KafkaSource || fail_test
+
+  uninstall_sources_crds || return 1
+  uninstall_channel_crds || return 1
+}
+
+# Note:  The setting of gcp-project-id option here has no effect when testing locally; it is only for the kubetest2 utility
+# If you wish to use this script just as test setup, *without* teardown, add "--skip-teardowns" to the initialize command
+initialize $@ --skip-istio-addon "--gcp-project-id=${PROJECT_ID}"
 
 export SYSTEM_NAMESPACE
 
-go_test_e2e -timeout=30m -parallel=12 ./test/e2e -channels=messaging.knative.dev/v1alpha1:KafkaChannel,messaging.knative.dev/v1beta1:KafkaChannel  || fail_test
+test_consolidated_channel || return 1
+
+# Terminate any zipkin port-forward processes that are still present on the system
+ps -e -o pid,command | grep 'kubectl port-forward zipkin[^*]*9411:9411 -n' | sed 's/^ *\([0-9][0-9]*\) .*/\1/' | xargs kill
 
-go_test_e2e -timeout=5m -parallel=12 ./test/conformance -channels=messaging.knative.dev/v1beta1:KafkaChannel -sources=sources.knative.dev/v1beta1:KafkaSource || fail_test
+test_distributed_channel || return 1
```

Comment:
I have the feeling that, on the long term, we should have 2 prow jobs, one for consolidated and one for distributed. This should be only a temp solution

Reply:
I don't think we need to duplicate test directories at the moment, we just need to set up different scripts

Reply:
I've modified the script to accept "--distributed" and/or "--consolidated" to test one or the other.  However, I'm not super-familiar with the test-infrastructure code that would run the custom prow jobs.  it it just a change to test-infra/config/prod/prow/config_knative.yaml to add two custom-test options to the knative-sandbox/eventing-kafka section (and remove the integration-tests entry)?  If so I can make those changes; if not I will need to ask for some help.  I seem to remember the way test-infra works changing recently.